xarray/util/generate_reductions.py

```diff
@@ -53,244 +51,327 @@ def reduce(
-        example=example_generator(obj=obj, method=method),
-    )
-
-
-def render(obj: str, cls: str, docref: str, example_generator: Callable):
-    yield CLASS_PREAMBLE.format(obj=obj, cls=cls)
-    yield generate_method(
-        obj,
-        method="count",
-        docref=docref,
-        skipna=False,
-        example_generator=example_generator,
-    )
-    for method in BOOL_REDUCE_METHODS:
-        yield generate_method(
-            obj,
-            method=method,
-            docref=docref,
-            skipna=False,
-            array_method=f"array_{method}",
-            example_generator=example_generator,
+            extra_kwargs = ""
+
+        yield TEMPLATE_REDUCTION_SIGNATURE.format(
+            **template_kwargs,
+            extra_kwargs=extra_kwargs,
         )
-    for method in NAN_REDUCE_METHODS:
-        yield generate_method(
-            obj,
-            method=method,
-            docref=docref,
-            skipna=True,
-            example_generator=example_generator,
+
+        for text in [
+            _DIM_DOCSTRING.format(method=method.name),
+            *(kwarg.docs for kwarg in method.extra_kwargs if kwarg.docs),
+            _KEEP_ATTRS_DOCSTRING,
+            _KWARGS_DOCSTRING.format(method=method.name),
+        ]:
+            if text:
+                yield textwrap.indent(text, 8 * " ")
+
+        yield TEMPLATE_RETURNS.format(**template_kwargs)
+
+        yield textwrap.indent(self.generate_example(method=method), "")
+
+        yield TEMPLATE_SEE_ALSO.format(
+            **template_kwargs,
+            docref=self.docref,
+            docref_description=self.docref_description,
         )
 
+        yield self.generate_code(method)
+
+    def generate_example(self, method):
+        create_da = f"""
+        >>> da = xr.DataArray({method.np_example_array}
+        ...     dims="time",
+        ...     coords=dict(
+        ...         time=("time", pd.date_range("01-01-2001", freq="M", periods=6)),
+        ...         labels=("time", np.array(["a", "b", "c", "c", "b", "a"])),
+        ...     ),
+        ... )"""
+
+        calculation = f"{self.datastructure.example_var_name}{self.example_call_preamble}.{method.name}"
+        if method.extra_kwargs:
+            extra_examples = "".join(
+                kwarg.example for kwarg in method.extra_kwargs if kwarg.example
+            ).format(calculation=calculation, method=method.name)
+        else:
+            extra_examples = ""
+
+        return f"""
+        Examples
+        --------{create_da}{self.datastructure.docstring_create}
+
+        >>> {calculation}(){extra_examples}"""
+
+    def generate_code(self, method):
+        extra_kwargs = [kwarg.call for kwarg in method.extra_kwargs if kwarg.call]
+
+        if self.datastructure.numeric_only:
+            extra_kwargs.append(f"numeric_only={method.numeric_only},")
+
+        if extra_kwargs:
+            extra_kwargs = "\n            " + "\n            ".join(extra_kwargs)
```

Comment:
Is there an alternative to writing `"\n        "` twice?

Reply:
`2 * "\n            "`?

Reply:
Ah but the `join` only applies to the second one...
```

Reply:
`new_line = "\n            "` ? Using a variable reduces the risk of typos at least.